File: kotlin/lesson 1/Shuffle.kt
// https://leetcode.com/problems/shuffle-an-array/

class Solution {
    fun shuffle(values: IntArray) {
        for (i in values.indices) {
            val r = (0 until i + 1).random()
            values[i] = values[r].also { values[r] = values[i] }
        }
    }
}

File: csharp/lesson 1/Shuffle.cs
// https://leetcode.com/problems/shuffle-an-array/

class Solution {    
    public void Shuffle(int[] values) {
        Random random = new Random();
        for (int i = 0; i < values.Length; i++) {
            int r = random.Next(i + 1);
            int temp = values[i];
            values[i] = values[r];
            values[r] = temp;
        }
    }
}
